zerver/tests/test_markdown.py

```diff
@@ -2978,3 +2979,19 @@ def test_curl_code_block_without_validation(self) -> None:
 
         result = processor.run(markdown_input)
         self.assertEqual(result, expected)
+
+
+class MarkdownEmojiTest(ZulipTestCase):
+    def test_all_emoji_match_regex(self) -> None:
+        non_matching_emoji = [
+            emoji
+            for codepoint in codepoint_to_name
+            if not POSSIBLE_EMOJI_RE.fullmatch(
+                emoji := "".join(chr(int(c, 16)) for c in codepoint.split("-"))
+            )
+        ]
+        self.assertEqual(
+            non_matching_emoji,
+            # numbers in boxes shouldn't be converted to emoji images, so this is fine
```

Comment:
@andersk do you feel okay about just doing this?

Reply:
Yeah. (Clarify “numbers in boxes” → “unqualified numbers in boxes” in the comment.)

Reply:
done!